src/main/java/com/datadoghq/datadog_lambda_java/Extension.java

```diff
@@ -0,0 +1,65 @@
+package com.datadoghq.datadog_lambda_java;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+class Extension {
+
+    private final static String AGENT_URL = "http://127.0.0.1:8124";
+    private final static String HELLO_PATH = "/lambda/hello";
+    private final static String FLUSH_PATH = "/lambda/flush";
+    private final static String EXTENSION_PATH = "/opt/extensions/datadog-agent";
+
+    public static boolean setup() {
+        boolean shouldUseExtension = false;
+        if(isExtensionRuning(EXTENSION_PATH)) {
+            DDLogger.getLoggerImpl().debug("Extension has been detected");
+            if(hitHelloRoute(AGENT_URL, HELLO_PATH)) {
+                shouldUseExtension = true;
+            } else {
+                DDLogger.getLoggerImpl().debug("Impossible to call the hello route");
```

Comment:
Impossible to call -> Could not call (throughout file)

"Impossible" would generally mean that there is no way this could ever happen, rather than there was an error this one time.

Reply:
got it! thanks 🙏